src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,86 @@
+package duke;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
+
+import duke.command.Command;
+import duke.exception.InvalidInstructionFormatException;
+import duke.exception.InvalidInstructionLengthException;
+import duke.exception.InvalidTaskIndexException;
+import duke.exception.MissingFieldException;
+import duke.exception.TaskDoneException;
+import duke.exception.UnknownInstructionException;
+import duke.logic.StorageManager;
+import duke.logic.TaskList;
+import duke.logic.UiManager;
+import duke.logic.UserInputParser;
+
+/**
+ * Represents a Duke Chat bot.
+ * It contains a <code>TaskList</code> to track the user's tasks,
+ * a <code>UIManager</code> to handle user interactions and
+ * a <code>StorageManager</code> to handle storing of data.
+ */
+public class Duke {
+    private final UiManager uiManager;
+    private final StorageManager storageManager;
+    private TaskList taskList;
+
+    public Duke() {
+        this.uiManager = new UiManager();
+        this.storageManager = new StorageManager(CommonString.DUKE_FILE_PATH.toString());
+        try {
+            this.taskList = new TaskList(storageManager.loadData());
```

Comment:
Perhaps you can think about shifting your taskList object to the StorageManager Class? Having your taskList in your main class could possibly complicate your manipulations of the items inside!